content/post/03-keepassxc-freedesktop-secret.md

```diff
@@ -0,0 +1,136 @@
+
+# Setting up within keepassxc
+
+First thing to do is enabling the `Secret Service Integration` on keepassxc itself, this can
+be done by clicking on `tools` in the bar on top, then `settings` and scrolling down the left bar
+and clicking on `Secret Service Integration`. Then tick
+`Enable KeepassXC Freedesktop.org Secret Service Integration`.
+
+After enabling the `Secret Service Integration`, you need to change the database settings so
+the entries of the database are exposed via the integration.
+
+First click on `Database` on the bar on top then on `Database Settings ...` and then click on
+`Secret Service Integration` and tick the option `Expose entries under this group:`, below the
+option there is a tree manager so you can pick only a part of the entries of the database to
+be exposed.
+
+To make sure it works, have your database unlocked (should already be for changing the setting
+above) and use `secret-tool(1)` to query for a secret on your database.
+
+> The key called 'secret' is [REDACTED] for obvious reasons
+
+```ini
+$ secret-tool search --unlock Path lichess
+[/org/freedesktop/secrets/collection/all/8cbc4c2af67a428a8b7859fdaf25881b]
+label = lichess
+secret = [REDACTED]
+created = 2019-09-06 17:31:10
+modified = 2019-10-30 01:18:16
+attribute.Title = lichess
+attribute.URL = https://lichess.org
+attribute.UserName = voidlinux1
+attribute.Path = /lichess
+attribute.Uuid = 8cbc4c2af67a428a8b7859fdaf25881b
+attribute.Notes = 
+```
+
+> PS: hit me up on lichess.org if you want to play :D
+
+# Experience thus far 
+
+Replacing my previous system-keyring provider, gnome-keyring, proved to be
+overall good but with some very annoying quirks.
+
+Benefits:
+
+- I don't need more shell code to properly initialize gnome-keyring
+- I don't need to wrestle around with PAM to properly initialize and unlock the keyring
+- The place where i store my secrets in my computer and everywhere else is the same
+
+Drawbacks:
+
+- I can't use my system-keyring to automatically unlock the database on login and unlock
+- There is no pop-up dialog like in gnome-keyring to unlock the database when someone queries
+ for secrets using the freedesktop API
+
+# Integrating with ssh-agent
+
+Speaking of which, since gnome-keyring was gone i needed to re-enable `ssh-agent(1)` so
+i can use my precious precious (that i dispose and replace a lot) ssh keys everywhere.
+
+I start `ssh-agent(1)` on my .profile, which is run every time i login with the following
+code:
+
+```sh
+if ! pgrep -x ssh-agent -u $(id -u) >/dev/null; then
+	# This sets SSH_AUTH_SOCK and SSH_AGENT_PID variables
+	eval "$(ssh-agent -s)"
+	export SSH_AUTH_SOCK SSH_AGENT_PID
+	cat > "$XDG_RUNTIME_DIR/ssh-agent-env" <<- __EOF__
+	export SSH_AUTH_SOCK=$SSH_AUTH_SOCK
+	export SSH_AGENT_PID=$SSH_AGENT_PID
+	__EOF__
+else
+	if [ -s "$XDG_RUNTIME_DIR/ssh-agent-env" ]; then
+		. $XDG_RUNTIME_DIR/ssh-agent-env
+	fi
+fi
+```
+
+It checks if there is an `ssh-agent(1)` running as me, if not then it starts
```

Comment:
s/as me/under my user ID/ ?

Reply:
Maybe also place a full stop there instead of a comma to make the sentence smaller